### DISCUSSION TITLE
Simulate realizations of a probabilistic forecast? Or other ways to propagate the confidence intervals to an aggregate of the forecast?

### DISCUSSION THREAD

**Discussion by emtilt** [alan-turing-institute/sktime#5549] [General]
I'm considering the case of a probabilistic forecast (at, say, a daily cadence) that I would like to aggregate (to, say, find the total of the forecasted values along with its confidence intervals, or to aggregate to a coarser sampling in time like weekly). My thinking (though I'm open to better suggestions!) for how to do this is to generate multiple realizations of the forecast that sample its distribution. The aggregated sum from each realization then samples the distribution of the desired sum. 

Is there an obvious way to do this from within `sktime`? For example, if I were to do this in `R` with the `forecast` package I could just use the `simulate` function. I'm having trouble locating anything related to this in the `sktime` documentation -- does a feature like this exist, or is there a way to access sufficient information about the model to write code to do this?

I figure something related to this must exist, but I can't find it, so I apologize if this question has an obvious answer!

**Comment by fkiraly** [comment]
For a programmatic way to simulate from a probabilistic forecast, you have to:

* first, predict a full distribution via `predict_proba`
* then, use the `sample` method

Note, that for most models, this will result in independent samples for each future forecast horizon, which is perhaps not what you are looking for. The framework would in-principle support distributions that are dependent across time indices, though these are either not implemented by the model, or integrated with such functionality in downstream packages that have it, e.g., `statsmodels`.

So far, this was not that highly requested, but I'd be happy to point you to where things need to be added.

Related issue here: https://github.com/sktime/sktime/issues/4098

**Reply by emtilt** [reply]
Thanks for the quick reply! Much appreciated!

I'm trying to understand what you mean by:

> this will result in independent samples for each future forecast horizon, which is perhaps not what you are looking for. The framework would in-principle support distributions that are dependent across time indices, though these are either not implemented by the model, or integrated with such functionality in downstream packages that have it, e.g., statsmodels.

I think you're saying that in general using the `sample` method would not work for my purposes, but I may be misunderstanding you so I just want to clarify to be sure. Are you saying that this method would draw samples from each time index in the forecast in such a way that they are independent of all other time indices, and thus there's not yet a way to simulate an actual timeseries realization with the usual property that the point corresponding to time index i is dependent on the point corresponding to time index i-1? So using the `sample` method as you mention to generate one complete forecast over the horizon would create points corresponding to i and i-1 that are independent of each other (and thus lacking one of the usual defining characteristics of a time series)? (For example, to make it concrete, if I'm forecasting daily data one week into the future, are you saying that there's not a way to generate a realization of that one week forecast such that the seventh day is dependent on the sixth?)

You also used this phrase:
> for most models

...which suggests that the `sample` method is implemented differently for different models. Can you expound on that?

**Reply by fkiraly** [reply]
> thus there's not yet a way to simulate an actual timeseries realization with the usual property that the point corresponding to time index i is dependent on the point corresponding to time index i-1?

Yes, for most forecasters that produce proba forecasts.

> ...which suggests that the `sample` method is implemented differently for different models. Can you expound on that?

Forecasters' `predict_proba` produces a predictive distribution - this can be a distribution that is independent across time indices, or not. Depending on what distribution is returned, the independence property that you want holds, or not.

**Reply by emtilt** [reply]
Is there a summary table, or other easy way to tell how `predict_proba` is handled by the various forecasters (since there's a lot!)?
